src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -664,51 +668,63 @@ static int fmi3_xml_str_to_array_floatXX(fmi3_xml_parser_context_t* context, con
 
     assert(str);
 
-    /* create a copy that it's OK that strtok mutates */
+    /* Check first that the current function has implemented support for the type passed to this function.
+    *   Otherwise we will have errors further down where check if the parsing of scalars/arrays is OK.
+    */
+    if (!(fmi3_base_type_enu_is_int(primType->baseType) ||
+          fmi3_base_type_enu_is_float(primType->baseType))) {
+        assert(0);
+    }
+
+    /* Create a copy that it's OK that strtok mutates */
     strCopy = context->callbacks->malloc(strlen(str) + 1); /* plus one for null character */
     if (!strCopy) {
         JM_LOG_ERROR_NO_MEM();
         res = -1;
-        goto err1;
+        goto strCopy_malloc_error;
```

Comment:
Please keep names `err1`, and only add new labels for cleanup i.e. freeing memory. It's kind of an "error stack", where you "push" a new label every time you allocate memory that needs to be freed on (local) failure.
See:
https://stackoverflow.com/questions/788903/valid-use-of-goto-for-error-management-in-c